### PR TITLE
Merge pull request #13 from Ab2nour:create-detrend-object

### PR DESCRIPTION
**Pull request by jExbrayat**
Create-detrend-object

### CODE REVIEW

File: data/processed_data/detrend_data/README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Detrend data folder
+
+It contains stocks price time series detrended with several methods and parameters.
```
Comment (Ab2nour) [inline data/processed_data/detrend_data/README.md]: nit: write "This folder contains"

File: data/processed_data/detrend_data/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+
+For each stock, open / close / low / high prices are detrended based on close price (chosen arbitrarily).
+Volumes aren't changed.
```
Comment (Ab2nour) [inline data/processed_data/detrend_data/README.md]: nit: "are not"

File: data/processed_data/detrend_data/README.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
+# Detrend data folder
+
+It contains stocks price time series detrended with several methods and parameters.
+
+For each stock, open / close / low / high prices are detrended based on close price (chosen arbitrarily).
```
Comment (Ab2nour) [inline data/processed_data/detrend_data/README.md]: nit: add comma "are detrended, based on close price"

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# docs folder
+
+This folder contains usefull documentations to understand the objective or technics used in the project.
```
Comment (Ab2nour) [inline docs/README.md]: typos: useful, documentation, techniques

File: src/functions/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# src/functions folder
+
+This folder contains python functions for the project, called in notebooks or data processing scripts.
```
Comment (Ab2nour) [inline src/functions/README.md]: nit: Python

File: src/functions/write_detrend_data.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import os
+import sys
+
+import pandas as pd
+
+sys.path.append("./src/functions")
+from detrend import LinearMA  # noqa: E402
+
+# SPECIFY DETREND MODEL AND DIRECTORY IN WHICH SAVE PROCESSED DATA
```
Comment (Ab2nour) [inline src/functions/write_detrend_data.py]: nit: no uppercase, write in lower case (or lowercase with first letter uppercased)

File: src/functions/write_detrend_data.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+detrend_model = LinearMA(window=200)
+model_name: str = "LinearMA"
+model_options: str = "window-201"  # format: option1-value_option2-value
+processed_data_folder = f"data/processed_data/detrend_data/{model_name}/{model_options}"
+
+
+# GATHER FILES TO PROCESS
```
Comment (Ab2nour) [inline src/functions/write_detrend_data.py]: nit: no uppercase, write in lower case (or lowercase with first letter uppercased)

File: src/functions/write_detrend_data.py [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+data_folder = "data/raw_data"
+filenames = [f for f in os.listdir(data_folder) if f.endswith(".csv")]
+
+# READ FILES
```
Comment (Ab2nour) [inline src/functions/write_detrend_data.py]: nit: no uppercase, write in lower case (or lowercase with first letter uppercased)

File: src/functions/write_detrend_data.py [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+
+data_files: dict = {
+    filename: pd.read_csv(
+        f"{data_folder}/{filename}", parse_dates=["Date"], index_col="Date"
+    )
+    for filename in filenames
+}
+
+# DETREND FILES
```
Comment (Ab2nour) [inline src/functions/write_detrend_data.py]: nit: no uppercase, write in lower case (or lowercase with first letter uppercased)

File: src/functions/write_detrend_data.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+    for data_type in data.columns[:-1]:  # data_type = Open / High etc. EXCEPT Volume
+        # load time series
+        y = data[data_type]
+        # detrend time series
+        y_predict = detrend_model.predict(y)
+        # overwrite original time series
+        data[data_type] = y_predict
+
+# WRITE PROCESSED FILES
```
Comment (Ab2nour) [inline src/functions/write_detrend_data.py]: nit: no uppercase, write in lower case (or lowercase with first letter uppercased)

File: src/notebooks/README.md [hunk 11]
```diff
@@ -1,6 +1,7 @@
-# src folder
+# src/notebooks folder
 
 This folder contains all the source code of the project.
 
 * `download-data.ipynb` is a notebook to download any stock data with yfinance.
 * `exploratory-data-analysis` is a notebook containing an EDA of the AAPL stock.
+* `decomposition.ipynb` is a notebook for reasearch in time series decomposition (trend, seasonanality, periodidicity, stochastic).
```
Comment (Ab2nour) [inline src/notebooks/README.md]: typo: periodicity

**Review by Ab2nour — CHANGES_REQUESTED** [review]
Nitpicking & fixing typos. 

Otherwise, there are improvements, but for the time being, LGTM.

File: data/processed_data/detrend_data/README.md (hunk 1 above)
Comment (jExbrayat) [reply data/processed_data/detrend_data/README.md]: done

File: data/processed_data/detrend_data/README.md (hunk 2 above)
Comment (jExbrayat) [reply data/processed_data/detrend_data/README.md]: done

File: data/processed_data/detrend_data/README.md (hunk 3 above)
Comment (jExbrayat) [reply data/processed_data/detrend_data/README.md]: done

File: docs/README.md (hunk 4 above)
Comment (jExbrayat) [reply docs/README.md]: done

File: src/functions/README.md (hunk 5 above)
Comment (jExbrayat) [reply src/functions/README.md]: done

File: src/functions/write_detrend_data.py (hunk 7 above)
Comment (jExbrayat) [reply src/functions/write_detrend_data.py]: done

File: src/functions/write_detrend_data.py (hunk 8 above)
Comment (jExbrayat) [reply src/functions/write_detrend_data.py]: done

File: src/functions/write_detrend_data.py (hunk 9 above)
Comment (jExbrayat) [reply src/functions/write_detrend_data.py]: done

File: src/functions/write_detrend_data.py (hunk 10 above)
Comment (jExbrayat) [reply src/functions/write_detrend_data.py]: done

File: src/notebooks/README.md (hunk 11 above)
Comment (jExbrayat) [reply src/notebooks/README.md]: done

File: src/functions/write_detrend_data.py (hunk 6 above)
Comment (jExbrayat) [reply src/functions/write_detrend_data.py]: done

**Review by Ab2nour — APPROVED** [review]

**Pull request merged**